spring-cloud-starter-stream-sink-log/src/main/java/org/springframework/cloud/stream/app/log/sink/LogSinkConfiguration.java

```diff
@@ -40,7 +47,20 @@
 	@Bean
 	@ServiceActivator(inputChannel = Sink.INPUT)
 	public LoggingHandler logSinkHandler() {
-		LoggingHandler loggingHandler = new LoggingHandler(this.properties.getLevel().name());
+		LoggingHandler loggingHandler = new LoggingHandler(this.properties.getLevel().name()) {
+
+			@Override
+			protected void handleMessageInternal(Message<?> message) throws Exception {
+				String contentType = message.getHeaders().containsKey(MessageHeaders.CONTENT_TYPE)
+						? message.getHeaders().get(MessageHeaders.CONTENT_TYPE).toString()
+						: BindingProperties.DEFAULT_CONTENT_TYPE.toString();
+				if (contentType.contains("text") || contentType.contains("json")) {
+					message = MessageBuilder.withPayload(new String(((byte[])message.getPayload()), StandardCharsets.UTF_8))
```

Comment:
Doesn't looks like you check the `payload` type.
What will happen if I have an `inputType` configured?
Is this `payload` still `byte[]`?

Thanks

Reply:
Artem, i think you meant `content-type` and if so, then as you and I discussed, there is no conversion strictly based on `content-type`. Only based on `type`. MessageConverters that do the conversion may also use content-type as a supplement info